### DISCUSSION TITLE
Server status code doesn't match with what async fetch api gets as response when using RSC

### DISCUSSION THREAD

**Discussion by jeepers3327** [vercel/next.js#48056] [Help]
### Summary

I'm trying it NextJS with the `/app` directory and use RSC. I encountered maybe a bug or some configuration I needed to do which I'm not aware of. The issue is that the server returns `401` status but fetch api returns a `200` status error. This is not the case for fetching through client which correctly returns the exepcted status code.

server clearly returns `401` error status
![image](https://user-images.githubusercontent.com/70475641/230529231-b2cb7e04-82ca-4751-8239-ecdb8d5715e1.png)

response from server side fetching
![Screenshot 2023-04-07 103309](https://user-images.githubusercontent.com/70475641/230528990-3c211f12-67f7-4b19-8fda-6be4eea135f0.png)

response from client side fetching
![image](https://user-images.githubusercontent.com/70475641/230529383-381afdc7-b9d5-4a8a-8dab-00696d656745.png)

Why the server side and client side fetching returns different status codes?

### Additional information
I use the same code (both for RSC and client side fetching) for fetching data that requires authenticated user.
```js


  try {
    const response = await fetch("http://localhost:4000/api/me", {
      headers: {
        "Content-Type": "application/json",
      },
      mode: "cors",
      credentials: "include",
    });

    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    console.log("status: " + response.status);
    console.log("is response ok: " + response.ok);
    return await response.json();
  } catch (error) {
    console.error(error);
  }
```

**Comment by jeepers3327** [comment]
After skimming through documentation again, I just need to add `cache: "no-store",` to fetch api options. Though I'm not sure if this is really the actual fix or not.

**Reply by MehakB7** [reply]
Hey did you able to fix this and find why it happened  ?
